core/src/test/java/org/dcache/testutils/InetAddressMockery.java

```diff
@@ -0,0 +1,108 @@
+package org.dcache.testutils;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectStreamConstants;
+import java.net.Inet4Address;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Creates InetAddress objects that contain arbitrary hostname/address combinations.
+ * 
+ * @author Christian Kohlschütter
+ */
+public final class InetAddressMockery {
+    private InetAddressMockery() {
+        throw new IllegalStateException("No instances");
+    }
+
+    /**
+     * Returns a "resolved" Inet4Address using the given hostname and IPv4 address.
+     * 
+     * @param hostname The hostname.
+     * @param ip4Address The IPv4 address (e.g. "1.2.3.4").
+     * @return The Inet4Address object.
+     * @throws IllegalStateException on internal error.
+     */
+    public static InetAddress mockedResolvedIPv4Address(String hostname, String ip4Address) {
+        byte[] addr;
+        try {
+            addr = ((Inet4Address) InetAddress.getByName(ip4Address)).getAddress();
+        } catch (UnknownHostException e) {
+            throw new IllegalStateException(e);
+        }
+        int ipv4Addr = ((addr[0]) << 24 & 0xFF000000) | (addr[1] << 16 & 0xFF0000)
+                | (addr[2] << 8 & 0xFF00) | (addr[3] & 0xFF);
+        return mockedResolvedIPv4Address(hostname, ipv4Addr);
+    }
+
+    /**
+     * Returns a "resolved" Inet4Address using the given hostname and IPv4 address.
+     * 
+     * @param hostname The hostname.
+     * @param ip4Address The IPv4 address (e.g. {@code 0x01020304})
+     * @return The Inet4Address object.
+     * @throws IllegalStateException on internal error.
+     */
+    public static InetAddress mockedResolvedIPv4Address(String hostname, int ip4Address) {
+        try (ObjectInputStream in = new ObjectInputStream(new ByteArrayInputStream(serializeIp4(hostname,
+                ip4Address)))) {
+            return (InetAddress) in.readObject();
```

Comment:
what about just use standard `InetAddress.html#getByAddress(java.lang.String,byte[])`?

Reply:
Absolutely. I didn't think too hard when copying code from another project. Thanks for the reminder!